pep/app/pods/components/power-select/journal/component.ts

```diff
@@ -0,0 +1,95 @@
+import Component from '@glimmer/component';
+import PowerSelectInfinityWithSearch from 'pep/pods/components/power-select-infinity/with-search/component';
+import Journal from 'pep/pods/journal/model';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import { DS } from 'ember-data';
+import { inject as service } from '@ember/service';
+import { didCancel } from 'ember-concurrency';
+import { restartableTask } from 'ember-concurrency-decorators';
+import FastbootService from 'ember-cli-fastboot/services/fastboot';
+
+interface JournalParams {
+    limit: number | null;
+    offset: number | null;
+    sourcecode?: any;
+}
+
+export default class PowerSelectJournal extends Component<PowerSelectInfinityWithSearch<Journal>> {
+    @service store!: DS.Store;
+    @service fastboot!: FastbootService;
+
+    @tracked options!: Journal[];
+    @tracked canLoadMore: boolean = true;
+    @tracked pageSize: number = 20;
+
+    /**
+     * Queries the server for `journals` using the passed in
+     * keyword.
+     *
+     * @param {string} keyword
+     * @param {number} offset
+     * @returns Journal[]
+     */
+    @restartableTask
+    *load(keyword: string, offset: number) {
+        try {
+            const params: JournalParams = {
+                limit: this.pageSize,
+                offset: offset || 0,
+                sourcecode: keyword
+            };
+            const result = yield this.store.query('journal', params);
+            let results = result.toArray();
+            this.canLoadMore = results.length >= this.pageSize;
+            return results;
+        } catch (errors) {
+            if (!didCancel(errors)) {
+                // this.notifications.groupErrors(errors);
+                throw errors;
+            }
+        }
+    }
+
+    /**
+     * Loads the initial page of options
+     * @return {Array}
+     */
+    @action
+    async loadInitialPage() {
+        if (!this.fastboot.isFastBoot) {
```

Comment:
use the `@dontRunInFastboot` decorator instead of this if block, and then i think you can remove the service injection itself too.